presto-hive/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java

```diff
@@ -396,6 +399,40 @@ public static RoleGrant fromRolePrincipalGrant(RolePrincipalGrant grant)
         return ImmutableSet.copyOf(result);
     }
 
+    public static Set<String> listEnabledRoles(Identity identity, Function<PrestoPrincipal, Set<RoleGrant>> listRoleGrants)
+    {
+        Optional<SelectedRole> role = identity.getRole();
+
+        if (role.isPresent() && role.get().getType() == SelectedRole.Type.NONE) {
+            return ImmutableSet.of("public");
+        }
+
+        PrestoPrincipal principal;
+        if (!role.isPresent() || role.get().getType() == SelectedRole.Type.ALL) {
+            principal = new PrestoPrincipal(USER, identity.getUser());
```

Comment:
Yes. And it is completely connector specific. Connector is chosing what is the default ;-)